CMakeLists.txt

```diff
@@ -692,30 +706,41 @@ endif()
 
 # Executable
 if (${PLAYER_BUILD_EXECUTABLE})
-	add_executable(${PROJECT_NAME}_exe WIN32 "src/main.cpp")
-	set_target_properties(${PROJECT_NAME}_exe PROPERTIES OUTPUT_NAME "easyrpg-player")
-
+	if(APPLE)
+		set(EXE_NAME Player.app)
+		set_source_files_properties(${${PROJECT_NAME}_BUNDLE_ICON} PROPERTIES MACOSX_PACKAGE_LOCATION "Resources")
+		add_executable(${EXE_NAME} MACOSX_BUNDLE "src/main.cpp" ${${PROJECT_NAME}_BUNDLE_ICON})
+		set_target_properties(${EXE_NAME} PROPERTIES MACOSX_BUNDLE_INFO_PLIST "resources/osx/Info.plist")
+		set_target_properties(${EXE_NAME} PROPERTIES OUTPUT_NAME "Player")
+	else()
+		set(EXE_NAME ${PROJECT_NAME}_exe)
+		add_executable(${EXE_NAME} WIN32 "src/main.cpp")
+		set_target_properties(${EXE_NAME} PROPERTIES OUTPUT_NAME "easyrpg-player")
+	endif()
+	
 	if(WIN32)
 		# Open console for Debug builds
 		if(CMAKE_BUILD_TYPE STREQUAL "Debug")
-			set_target_properties(${PROJECT_NAME}_exe PROPERTIES WIN32_EXECUTABLE FALSE)
+			set_target_properties(${EXE_NAME} PROPERTIES WIN32_EXECUTABLE FALSE)
 		endif()
 
 		# Add resources
-		target_sources(${PROJECT_NAME}_exe PRIVATE "resources/player.rc")
+		target_sources(${EXE_NAME} PRIVATE "resources/player.rc")
 
 		# Change executable name
-		set_target_properties(${PROJECT_NAME}_exe PROPERTIES OUTPUT_NAME "Player")
+		set_target_properties(${EXE_NAME} PROPERTIES OUTPUT_NAME "Player")
 	endif()
 
-	target_link_libraries(${PROJECT_NAME}_exe ${PROJECT_NAME})
+	target_link_libraries(${EXE_NAME} ${PROJECT_NAME})
 
-	# installation
-	include(GNUInstallDirs)
-	install(TARGETS ${PROJECT_NAME}_exe RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+	# installation - don't use install on macs
```

Comment:
Did this because I'm getting a weird error I don't know how to fix, and I'm also not entirely sure where cmake would put a `.app` if it were to install it...